<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE TestGroup SYSTEM "testgenspec.dtd">
<!--

    Copyright (c) 2018, 2020 Oracle and/or its affiliates. All rights reserved.

    This program and the accompanying materials are made available under the
    terms of the Eclipse Public License v. 2.0, which is available at
    http://www.eclipse.org/legal/epl-2.0.

    This Source Code may also be made available under the following Secondary
    Licenses when the conditions for such availability set forth in the
    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
    version 2 with the GNU Classpath Exception, which is available at
    https://www.gnu.org/software/classpath/license.html.

    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0

-->

<TestGroup ID="ValidationException">
    <TestGroupDocumentation>
        <Title><![CDATA[General tests of constructors and methods]]></Title>
        <Description><![CDATA[General tests of constructors and methods.]]></Description>
        <TestedPackage>jakarta.xml.bind</TestedPackage>
        <TestedClass>ValidationException</TestedClass>
        <Author><![CDATA[Evgueni Rouban]]></Author>
    </TestGroupDocumentation>

    <TestGroupAttributes>
       <AttrElem Name="tdFile">ValidationException</AttrElem>
       <AttrElem Name="scInfo">@(#)ValidationException.test.xml	1.1 02/11/11</AttrElem>
       <AttrElem Name="testType">MultiTest</AttrElem>
       <Keyword>runtime</Keyword>
       <Keyword>positive</Keyword>
    </TestGroupAttributes>

    <CodeSet>
        <Import>jakarta.xml.bind.ValidationException</Import>
        <SupportCode><![CDATA[
    static final String msg = "test message";
    static final String err = "error code";
    static final Throwable linkedException = new Throwable();
]]></SupportCode>
    </CodeSet>

    <TestCase ID="Ctor001">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified detail message. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message)</MemberSig>
                <ExpectedResultSideEffect>the message is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg);
    if (ve.getMessage() != msg) {
        return Status.failed("getMessage() != msg: \"" 
                           + ve.getMessage() + "\", expected is \"" + msg + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor002">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
The errorCode will default to null.
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message)</MemberSig>
                <ExpectedResultSideEffect>errorCode is null</ExpectedResultSideEffect>
            </TestCaseSpec>
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg);
    if (ve.getErrorCode() != null) {
        return Status.failed("getErrorCode() != null: \"" 
                           + ve.getErrorCode() + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor003">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
The linkedException will default to null.
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message)</MemberSig>
                <ExpectedResultSideEffect>linkedException is null</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg);
    if (ve.getLinkedException() != null) {
        return Status.failed("getLinkedException() != null: \"" 
                           + ve.getLinkedException() + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor004">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified detail message. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.String errorCode)</MemberSig>
                <ExpectedResultSideEffect>the message is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, err);
    if (ve.getMessage() != msg) {
        return Status.failed("getMessage() != msg: \"" 
                           + ve.getMessage() + "\", expected is \"" + msg + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor005">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified errorCode. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.String errorCode)</MemberSig>
                <ExpectedResultSideEffect>the errorCode is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, err);
    if (ve.getErrorCode() != err) {
        return Status.failed("getErrorCode() != err: \"" 
                           + ve.getErrorCode() + "\", expected is \"" + err + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor006">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
The linkedException will default to null.
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.String errorCode)</MemberSig>
                <ExpectedResultSideEffect>linkedException is null</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, err);
    if (ve.getLinkedException() != null) {
        return Status.failed("getLinkedException() != null: \"" 
                           + ve.getLinkedException() + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor007">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified detail message. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.String errorCode, java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>the message is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, err, linkedException);
    if (ve.getMessage() != msg) {
        return Status.failed("getMessage() != msg: \"" 
                           + ve.getMessage() + "\", expected is \"" + msg + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor008">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified errorCode. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.String errorCode, java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>the errorCode is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, err, linkedException);
    if (ve.getErrorCode() != err) {
        return Status.failed("getErrorCode() != err: \"" 
                           + ve.getErrorCode() + "\", expected is \"" + err + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor009">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified linkedException. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.String errorCode, java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>linkedException is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, err, linkedException);
    if (ve.getLinkedException() != linkedException) {
        return Status.failed("getLinkedException() != linkedException: \"" 
                           + ve.getLinkedException() + "\", expected is \"" + linkedException + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor010">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified detail message. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>the message is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, linkedException);
    if (ve.getMessage() != msg) {
        return Status.failed("getMessage() != msg: \"" 
                           + ve.getMessage() + "\", expected is \"" + msg + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor011">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
The errorCode will default to null.
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>errorCode is null</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, linkedException);
    if (ve.getErrorCode() != null) {
        return Status.failed("getErrorCode() != null: \"" 
                           + ve.getErrorCode() + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor012">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified linkedException. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.String message, java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>linkedException is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(msg, linkedException);
    if (ve.getLinkedException() != linkedException) {
        return Status.failed("getLinkedException() != linkedException: \"" 
                           + ve.getLinkedException() + "\", expected is \"" + linkedException + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor013">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
The detail message will default to null.
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>the message is null</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(linkedException);
    if (ve.getMessage() != null) {
        return Status.failed("getMessage() != null: \"" 
                           + ve.getMessage() + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor014">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
The errorCode will default to null.
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>errorCode is null</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(linkedException);
    if (ve.getErrorCode() != null) {
        return Status.failed("getErrorCode() != null: \"" 
                           + ve.getErrorCode() + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

    <TestCase ID="Ctor015">
        <TestCaseDocumentation>
            <TestCaseSpec>
                <InlineAssertion><![CDATA[
Construct a ValidationException with the specified linkedException. 
                 ]]></InlineAssertion>
                <MemberSig>public ValidationException(java.lang.Throwable exception)</MemberSig>
                <ExpectedResultSideEffect>linkedException is what set</ExpectedResultSideEffect>
            </TestCaseSpec>
            
        </TestCaseDocumentation>
        <TestCaseAttributes/>
        <TestCode><![CDATA[
    ValidationException ve = new ValidationException(linkedException);
    if (ve.getLinkedException() != linkedException) {
        return Status.failed("getLinkedException() != linkedException: \"" 
                           + ve.getLinkedException() + "\", expected is \"" + linkedException + "\"");
    }

    return Status.passed("OKAY");
]]></TestCode>
    </TestCase>

</TestGroup>
<!-- Ident: @(#)ValidationException.test.xml	1.1 02/11/11 -->
